onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -284,7 +290,14 @@ def add_reduce_min_max(tensor_name, reduce_op_name):
             )
 
             self.model.graph.node.extend([reduce_node, reshape_node])
-            self.model.graph.output.append(helper.make_tensor_value_info(reduce_output, TensorProto.FLOAT, [1]))
+            value_infos = {vi.name: vi for vi in self.model.graph.value_info}
+            value_infos.update({o.name: o for o in self.model.graph.output})
+            value_infos.update({i.name: i for i in self.model.graph.input})
+            if tensor_name in value_infos:
+                onnx_type = value_infos[tensor_name].type.tensor_type.elem_type
+            else:
+                raise ValueError(f"Unable to guess tensor type for tensor {tensor_name!r}")
```

Comment:
>raise ValueError(f"Unable to guess tensor type for tensor {tensor_name!r}") [](http://example.com/codeflow?start=16&length=75)

could you please add info to ask user to run shape infer before quantization in this case.

Reply:
I did.